MOBT-2-Swift-Foundation.playground/Contents.swift

```diff
@@ -0,0 +1,332 @@
+                                               comment: comment ?? "")
+        localizedValue = localizedValue.replacingOccurrences(of: "%s", with: "%@")
+        
+        return localizedValue
+    }
+    
+    /// Gets localized string for `format` and passes arguments into the string
+    func localizedString(format: String, _ arguments: CVarArg...) -> String {
+        let localizedFormat = localizedString(format)
+        return String(format: localizedFormat, arguments: arguments)
+    }
+}
+
+
+class Configurator {
+    static var currentTarget: Target = {
+        guard let targetName = Bundle.main.infoDictionary?["TargetName"] as? String,
+              let target = Target(rawValue: targetName) else {
+                  fatalError("Cannot find current target name")
+              }
+        return target
+    }()
+}
+/// This peace of code we use to separate some kind of countries for different targets
+enum Target: String {
+    case Italy = "Italy"
+    case Greece = "Greece"
+    case Romania = "Romania"
+    case Netherlands = "Netherlands"
+    
+    var configurationPath: String {
+        switch self {
+        case .Italy:
+            return "ConfigurationItaly"
+        case .Greece:
+            return "ConfigurationGreece"
+        case .Romania:
+            return "ConfigurationRomania"
+        case .Netherlands:
+            return "ConfigurationNetherlands"
+        }
+    }
+    
+    var localizedTableName: String {
+        switch self {
+        case .Italy:
+            return "LocalizedItaly"
+        case .Greece:
+            return "LocalizedGreece"
+        case .Romania:
+            return "LocalizedRomania"
+        case .Netherlands:
+            return "ConfigurationNetherlands"
+        }
+    }
+}
+
+/// This is the example of getting backend part of code. And transfer this data to swift objects.
+// MARK: - Typealias
+public typealias EmptyClosure = () -> Void
+public typealias DataKeys = (([String]) -> Void)
+public typealias DataValues = (([Double]) -> Void)
+
+public struct Keys {
+    static let mostRecentExchangeRatesValues = "Keys.mostRecentExchangeRatesValue"
+    static let mostRecentExchangeRatesKeys = "Keys.mostRecentExchangeRatesKeys"
+}
+
+// MARK: - Protocols
+protocol CurrencyExchangeMethods {
+    func fetchJSON(retreiveDataKeys: @escaping DataKeys, retrieveDataValues: @escaping DataValues)
+}
+
+// MARK: - Classes
+final class CurrencyExchangeConverterService: CurrencyExchangeMethods {
+    // MARK: - Enums
+    private enum Const {
+        static let urlAPI: String = "https://open.er-api.com/v6/latest/USD"
+    }
+    
+    // MARK: - Properties
+    var currencyCode: [String] = []
+    var values: [Double] = []
+    
+    // MARK: - Methods
+    func fetchJSON(retreiveDataKeys: @escaping DataKeys, retrieveDataValues: @escaping DataValues) {
+        guard let url = URL(string: Const.urlAPI) else { return }
+        URLSession.shared.dataTask(with: url) { (data, response, error) in
```

Comment:
We should avoid using shared state:
1) If session is needed — it should be injected via the initialiser
2) Probably it's a good idea to cover URLSession instance with a protocol